unity/Assets/Scripts/BaseFPSAgentController.cs

```diff
@@ -4501,4 +4526,17 @@ float allowedError
         }
 
     }
+    
+    public class VisibilityCheck {
+        public bool visible;
+        public bool interactable;
+
+        public static VisibilityCheck operator |(VisibilityCheck a, VisibilityCheck b) {
+            VisibilityCheck c = new VisibilityCheck();
+            c.interactable = a.interactable | b.interactable;
+            c.visible = a.visible | b.visible;
```

Comment:
Also we should probably replace `|` with `||` in the below?
```c#
            c.interactable = a.interactable | b.interactable;
            c.visible = a.visible | b.visible;
```

Reply:
yup!